front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts

```diff
@@ -0,0 +1,136 @@
+import type { WithAPIErrorResponse } from "@dust-tt/types";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import {
+  deleteDataSource,
+  MANAGED_DS_DELETABLE_AS_BUILDER,
+} from "@app/lib/api/data_sources";
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { DataSourceResource } from "@app/lib/resources/data_source_resource";
+import { VaultResource } from "@app/lib/resources/vault_resource";
+import { apiError } from "@app/logger/withlogging";
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<void>>,
+  auth: Authenticator
+): Promise<void> {
+  const owner = auth.workspace();
+  const plan = auth.plan();
+  const user = auth.user();
+
+  if (!owner || !plan || !user) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you requested was not found.",
+      },
+    });
+  }
+
+  const vault = await VaultResource.fetchById(auth, req.query.vId as string);
+
+  if (!vault || !auth.hasPermission([vault.acl()], "write")) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "vault_not_found",
+        message: "The vault you requested was not found.",
+      },
+    });
+  }
+
+  if (vault.isSystem() && !auth.isAdmin()) {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message:
+          "Only the users that are `admins` for the current workspace can update a data source.",
+      },
+    });
+  } else if (vault.isGlobal() && !auth.isBuilder()) {
+    return apiError(req, res, {
+      status_code: 403,
+      api_error: {
+        type: "data_source_auth_error",
+        message:
+          "Only the users that are `builders` for the current workspace can update a data source.",
+      },
+    });
+  } else if (!auth.isUser()) {
```

Comment:
Can this case happens? I feel like checking `auth.user` at the beginning like it's currently done should be enough, right?